c/datasetjson.c

```diff
@@ -2517,6 +2525,391 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
+        }
+      } else if(!strcmp(propString, "lrecl")) {
+        long toi = strtol(valueString, NULL, 0);
+        if (errno != ERANGE){
+          if (toi == 0x8000 || (toi <= 0x7FF8 && toi >= 0)) {
+            rc = setTextUnit(TEXT_UNIT_INT16, 0, NULL, toi, DALLRECL, configsCount, inputTextUnit);
+          }
+        }
+      } else if(!strcmp(propString, "volser")) {
+        if (valueStrLen <= VOLSER_SIZE){
+          rc = setTextUnit(TEXT_UNIT_STRING, VOLSER_SIZE, &(valueString)[0], 0, DALVLSER, configsCount, inputTextUnit);
+        }
+      } else if(!strcmp(propString, "recfm")) {
+        int setRECFM = 0;
+        if (indexOf(valueString, valueStrLen, 'A', 0) != -1){
+          setRECFM = setRECFM | DALRECFM_A;
+        }
+        if (indexOf(valueString, valueStrLen, 'B', 0) != -1){
+          setRECFM = setRECFM | DALRECFM_B;
+        }
+        if (indexOf(valueString, valueStrLen, 'V', 0) != -1){
+          setRECFM = setRECFM | DALRECFM_V;
+        }
+        if (indexOf(valueString, valueStrLen, 'F', 0) != -1){
+          setRECFM = setRECFM | DALRECFM_F;
+        }
+        if (indexOf(valueString, valueStrLen, 'U', 0) != -1){
+          setRECFM = setRECFM | DALRECFM_U;
+        }
+        rc = setTextUnit(TEXT_UNIT_CHAR, 0, NULL, setRECFM, DALRECFM, configsCount, inputTextUnit);
+      } else if(!strcmp(propString, "blkln")) {
+        long toi = strtol(valueString, NULL, 0);
+        if (errno != ERANGE) {
+          if (toi <= 0xFFFF || toi >= 0){
+            rc = setTextUnit(TEXT_UNIT_INT24, 0, NULL, toi, DALBLKLN, configsCount, inputTextUnit);
+          }
+        }
+      } else if(!strcmp(propString, "status")) {
+        if (!strcmp(valueString, "OLD")){
+          parmDefn = DISP_OLD;
+        } else if (!strcmp(valueString, "MOD")){
+          parmDefn = DISP_MOD;
+        } else if (!strcmp(valueString, "SHARE")){
+          parmDefn = DISP_SHARE;
+        } else {
+          parmDefn = DISP_NEW;
+        }
+        rc = setTextUnit(TEXT_UNIT_CHAR, 0, NULL, parmDefn, DALSTATS, configsCount, inputTextUnit);
+      } else if (!strcmp(propString, "ndisp")) {
+        if (!strcmp(valueString, "UNCATLG")){
+          parmDefn = DISP_UNCATLG;
+        } else if (!strcmp(valueString, "DELETE")){
+          parmDefn = DISP_DELETE;
+        } else if (!strcmp(valueString, "KEEP")){
+          parmDefn = DISP_KEEP;
+        } else {
+          parmDefn = DISP_CATLG;
+        }
+        rc = setTextUnit(TEXT_UNIT_CHAR, 0, NULL, parmDefn, DALNDISP, configsCount, inputTextUnit);
+      } else if(!strcmp(propString, "unit")) {
+        rc = setTextUnit(TEXT_UNIT_STRING, valueStrLen, &(valueString)[0], 0, DALUNIT, configsCount, inputTextUnit);
+      } else if(!strcmp(propString, "spgnm")) {
+        if (valueStrLen <= CLASS_WRITER_SIZE){
+          rc = setTextUnit(TEXT_UNIT_STRING, valueStrLen, &(valueString)[0], 0, DALSPGNM, configsCount, inputTextUnit);
+        }
+      } else if(!strcmp(propString, "close")) {
```

Comment:
meaning "autounallocate"

Reply:
we should have code comments linking over here because these are super confusing and we will forget what these mean.
https://www.ibm.com/docs/en/zos/2.1.0?topic=mvs-zos-programming-authorized-assembler-services-guide
https://www-40.ibm.com/servers/resourcelink/svc00100.nsf/pages/zOSV2R3sa231371/$file/ieaa800_v2r3.pdf

Reply:
Do you want me to add "autounallocate" comment here? 
I have referred these docs https://www.ibm.com/docs/en/zos/2.4.0?topic=units-unallocation-close-specification-key-001c